saleor/graphql/order/mutations/draft_order_complete.py

```diff
@@ -98,33 +106,40 @@ def perform_mutation(cls, _root, info, id):
             channel = order.channel
             order_lines_info = []
             for line in order.lines.all():
-                if line.variant.track_inventory or line.variant.is_preorder_active():
-                    line_data = OrderLineInfo(
-                        line=line, quantity=line.quantity, variant=line.variant
-                    )
-                    order_lines_info.append(line_data)
-                    site = get_site_promise(info.context).get()
-                    try:
-                        with traced_atomic_transaction():
-                            allocate_stocks(
-                                [line_data],
-                                country,
-                                channel,
-                                manager,
-                                check_reservations=is_reservation_enabled(
-                                    site.settings
-                                ),
-                            )
-                            allocate_preorders(
-                                [line_data],
-                                channel.slug,
-                                check_reservations=is_reservation_enabled(
-                                    site.settings
-                                ),
+                if line.variant:
+                    # we only care about stock for variants that still exist
+                    if (
+                        line.variant.track_inventory
+                        or line.variant.is_preorder_active()
+                    ):
+                        line_data = OrderLineInfo(
+                            line=line, quantity=line.quantity, variant=line.variant
+                        )
+                        order_lines_info.append(line_data)
+                        site = get_site_promise(info.context).get()
+                        try:
+                            with traced_atomic_transaction():
+                                allocate_stocks(
```

Comment:
Could make this block less deep using `continue`

Reply:
Fixed